src/main/java/com/zendesk/maxwell/MaxwellContext.java

```diff
@@ -146,6 +150,21 @@ public Long getServerID() throws SQLException {
 	}
 
 
+	private void fetchMysqlVersion() throws SQLException {
+		if ( mysqlMajorVersion == null ) {
+			try ( Connection c = getReplicationConnection() ) {
+				DatabaseMetaData meta = c.getMetaData();
+				mysqlMajorVersion = meta.getDatabaseMajorVersion();
+				mysqlMinorVersion = meta.getDatabaseMinorVersion();
+			}
+		}
+	}
+
+	public boolean shouldHeartbeat() throws SQLException {
+		fetchMysqlVersion();
```

Comment:
it is already, see the `null` check on line 154.  inlining should taking care of the rest...